### PR TITLE
Added article about redis pub/sub scaling

### PR DESCRIPTION
**Pull request by steve-hb**

### CODE REVIEW

File: blog/technical/articles/redis-pub-sub.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+## The problem with scaling and the cloud
+
+This project was meant to be deployed to the cloud, but what has this to do with this topic?<br>
+The cloud allows us to start and stop instances of our micro services in just a few seconds.
+
+This clearly is a great advantage of the cloud, but what if we want to communicate between different instances? In reality it's hard to know the IP of every currently running micro service to try to communicate with them.<br>
+Micro services in a cloud start and stop (sometimes crash) without any interaction from our side.
+
+
+We also have the problem that not every instance wants to know everything currently happening in our infrastructure. Sending notifications to these micro services would mean alot of overhead which increases dramastically every time our infrastructure scales itself up.
```
Comment (TimyStream) [inline blog/technical/articles/redis-pub-sub.md]: Correction = "alot" -> "a lot"
Correction = "dramastically" -> "dramatically"

File: blog/technical/articles/redis-pub-sub.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+Because of that we need a _"central"_ instance knowing:
+
+* The IPs of every micro service and
+* What every instance really wants to know
+
+So, we need another micro service, right? No.<br>
+A single micro service handling all the things we want to share with other micro services, would mean we have to develop and maintain it.<br>
+Even if we would maintain it: What happens if the service itself crashes or gets slow because of millions of messages per minute? This would mean that we wouldn't be able to communicate between our micro services which could lead to 1. data loss and 2. unavailable features or even 3. broken security mechanisms.
```
Comment (TimyStream) [inline blog/technical/articles/redis-pub-sub.md]: Correction = "1. data" -> "1. Data"
Correction = "2. unavailable" -> "2. Unavailable"
Correction = "3. broken" -> "3. Broken"

Is this better ?

File: blog/technical/articles/redis-pub-sub.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+The Redis server now gets a message with a specific _topic_ from a **publisher** and distributes it to every **subscriber** in our network that has subscribed to the _topic_ before.
+
+That's the general concept of the Redis PubSub service. Now let's face new problems.
+
+### Central instances and cloud scaling
+
+A single Redis PubSub instance is fast, extremely fast, but even Redis has limitations in terms of the amount of messages per second it can handle.
+
+When desinging cloud infrastructure it's very important to (almost) never limitate your system: Even if you think you won't need that much performance the next few year, what is your plan after these few years? Your infrastructure is built and deployed which means changing something is a huge task and risk for your business.
```
Comment (TimyStream) [inline blog/technical/articles/redis-pub-sub.md]: Correction = "the next few year," -> "the next few years"

File: blog/technical/articles/redis-pub-sub.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+I wish I could write something like: Redis supports clustering - solution to all our problems!<br>
+But the (current) truth is: Redis Clusters are not made for pub/sub which means we would get **linear decreasing** performance when increasing the cluster size.
+
+So, why chose Redis nevertheless? There are architectural workarounds for Redis which allow us to scale our infrastructure without hitting limitations: [This talk is about how one company acheived this](https://youtu.be/6G22a5Iooqk)<br>
```
Comment (TimyStream) [inline blog/technical/articles/redis-pub-sub.md]: Correction = "This talk is about how one company acheived this" -> "This talk is about how one company achieved this"

**Review by TimyStream — CHANGES_REQUESTED** [review]
I would Change these for better Understanding an some Spelling Correction

File: blog/technical/articles/redis-pub-sub.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+The Redis server now gets a message with a specific _topic_ from a **publisher** and distributes it to every **subscriber** in our network that has subscribed to the _topic_ before.
+
+That's the general concept of the Redis PubSub service. Now let's face new problems.
+
+### Central instances and cloud scaling
+
+A single Redis PubSub instance is fast, extremely fast, but even Redis has limitations in terms of the amount of messages per second it can handle.
+
+When desinging cloud infrastructure it's very important to (almost) never limitate your system: Even if you think you won't need that much performance the next few years, what is your plan after these few years? Your infrastructure is built and deployed which means changing something is a huge task and risk for your business.
```
Comment (TimyStream) [inline blog/technical/articles/redis-pub-sub.md]: "never limitate your system"

Are you sure that this is right ?

Comment (steve-hb) [reply blog/technical/articles/redis-pub-sub.md]: I could replace it with "never limitate your infrastructure"

Comment (steve-hb) [reply blog/technical/articles/redis-pub-sub.md]: or "scaleability"

Comment (TimyStream) [reply blog/technical/articles/redis-pub-sub.md]: Replace it if you want to.

**Pull request merged**